doc/man1/flux-job.rst

```diff
@@ -59,6 +61,22 @@ selected with:
 **-q, --quiet**
    Suppress output if no jobs match
 
+WAIT
+====
+
+A waitable job may be waited on with ``flux job wait``.  A specific
+job can be waited on by specifying a jobid.  If no jobid is specified,
+the command will wait for any waitable user job to complete,
+outputting that jobid before exiting.  This command will exit with
+error if the job is not successful.
```

Comment:
hmmm I was just dumbly following what was said in #4768, but should we consider not documenting `flux job wait` and just document `flux job status`?  For the average user on the command line, I don't think they're going to care?

IIRC w/ some commit log skims, `wait` is the faster scalable version since it works off `job-manager` and not watching the KVS?  Average power user is going to use the equivalent python api call for wait?

hmmm, but I guess `flux job wait`'s advantage would be the `--all` option if its a lot of jobs, but they might use `flux queue drain/idle` instead.  Or possibly with a future `--any` option in #4769 

Just trying to think about how to document the difference, because they would appear identical at a glance.

Reply:
I think the reason that #4678 was opened was because questions about how to efficiently wait for many jobs has come up many times and we'd like to have a documented reference for `flux job wait` and when it can and can't be used.

@garlick should probably comment here, but we should say something to the effect that waitable jobs and `flux job wait` is the most efficient way to wait for job completion and is modeled after the shell `wait` command. It avoids use of the KVS, and should be used if possible for extremely performance critical cases. However, the caveats are that the job must have been submitted with the waitable flag, and this is only allowed for the instance owner. Also, similar to `waitpid(2)`, a job can only have `flux job wait` called on it once.

If any of these caveats presents a problem, then `flux job status ID...` may be used to fetch the final job status. This command is less efficient than `flux job wait` since it must access the KVS to determine the job status, but it has been
tested up to many thousands of jobs. Also, `flux job status` can be called on a single job multiple times if that is a requirement.

Reply:
Maybe a FAQ entry on this subject would be warranted as well.

> hmmm, but I guess flux job wait's advantage would be the --all option if its a lot of jobs, but they might use flux queue drain/idle instead. Or possibly with a future --any option in https://github.com/flux-framework/flux-core/issues/4769

I was thinking of a use case where a script submits many jobs then waits for them all at the end. Then a single instance could run many of these scripts, but that won't work because `flux job wait --all` waits for all waitable jobs in the instance, not just those submitted from a given script like the shell's `wait` builtin would :thinking: So in this case you'd have to collect all the relevant jobids in a variable and use `flux job wait $id` repeatedly, or `flux job status $jobids` (`flux job wait` doesn't seem to support multiple jobids, maybe that can be fixed)

I'm also wondering if `waitstatus` would have been a better name for `flux job status`. Sorry, getting off in the weeds now.